src/System.IO.FileSystem/tests/File/ReadWriteAllBytes.cs

```diff
@@ -114,5 +114,31 @@ public void WriteToReadOnlyFile_UnauthException()
             Assert.Throws<UnauthorizedAccessException>(() => File.WriteAllBytes(path, Encoding.UTF8.GetBytes("text")));
             File.SetAttributes(path, FileAttributes.Normal);
         }
+
+        /// <summary>
+        /// On Unix, modifying a file that is ReadOnly will fail under normal permissions.
+        /// If the test is being run under the superuser, however, modification of a ReadOnly
+        /// file is allowed. Since we don't know the user running these tests, we attempt
+        /// the operation and verify the result for either case.
+        /// </summary>
+        [Fact]
+        [PlatformSpecific(PlatformID.AnyUnix)]
+        public void Unix_WriteToReadOnlyFile()
+        {
+            string path = GetTestFilePath();
+            File.Create(path).Dispose();
+            File.SetAttributes(path, FileAttributes.ReadOnly);
+
+            try
+            {
+                File.WriteAllBytes(path, Encoding.UTF8.GetBytes("text"));
+                Assert.Equal(Encoding.UTF8.GetBytes("text"), File.ReadAllBytes(path));
+            }
```

Comment:
Yes, now that we've split the files into Windows/Unix we can add a geteuid check to the Unix tests. That will solve the Linux case, but I'm not sure what is happening specifically on OSX to be raising errors. I've been working under the assumption that the CI tests are running as the superuser on OSX. If that is incorrect, adding an ```Assert.Equal(0, getuid())``` call would not solve the failures that #4605 found.